### ISSUE TITLE
System Error when running Ganache 2.0.2-beta.0 on darwin

### ISSUE THREAD

**Issue by hectorchu** [trufflesuite/ganache#1333]
PLATFORM: darwin
GANACHE VERSION: 2.0.2-beta.0

EXCEPTION:
```
UnknownBaseContractIdError: Cannot locate base contract ID 1229$ of contract$ TestDapp (ID 1215)
    at vars.concat.linearizedBaseContractsFromBase.map (/node_modules/truffle-decoder/dist/allocate/storage.js:158:19)
    at Array.map (native)
    at allocateContract (/node_modules/truffle-decoder/dist/allocate/storage.js:155:61)
    at Object.getStorageAllocations (/node_modules/truffle-decoder/dist/allocate/storage.js:29:23)
    at TruffleContractDecoder.<anonymous> (/node_modules/truffle-decoder/dist/interface/contract-decoder.js:113:47)
    at Generator.next (<anonymous>)
    at fulfilled (/node_modules/truffle-decoder/dist/interface/contract-decoder.js:4:58)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

APPLICATION LOG:
```
T+16048ms:   Contract created: 0x380e6a75a3f60ccd8214d2948e2b9e8090e10105
T+16048ms:   Gas usage: 261393
T+16079ms:   Block Number: 1
T+16079ms:   Block Time: Mon Jun 24 2019 07:22:34 GMT+0100 (BST)
T+16109ms: eth_getBlockByNumber
T+16131ms: eth_getTransactionReceipt
T+16157ms: eth_getCode
T+16185ms: eth_getTransactionByHash
T+16213ms: eth_getBlockByNumber
T+16233ms: eth_getBalance
T+16264ms: eth_getBlockByNumber
T+16293ms: eth_getBlockByNumber
T+16322ms: eth_sendTransaction
T+16352ms:   Transaction: 0xaee742bfd2997f5534fbece46dee9f9f58176f0397adefb1bc8b7ccc438c84c2
T+16352ms:   Gas usage: 42023
T+16352ms:   Block Number: 2
T+16352ms:   Block Time: Mon Jun 24 2019 07:22:35 GMT+0100 (BST)
T+16373ms: eth_getBlockByNumber
T+16401ms: eth_getTransactionReceipt
T+16430ms: eth_getBlockByNumber
T+16457ms: eth_accounts
T+16477ms: eth_getBlockByNumber
T+16507ms: eth_getBlockByNumber
T+16616ms: eth_getBlockByNumber
T+16616ms: eth_estimateGas
T+16616ms: eth_getBlockByNumber
T+16616ms: eth_blockNumber
T+16616ms: eth_sendTransaction
T+16634ms:   Transaction: 0x05e5fb2d827d01c4d6f1f20cd66c472c504c3b8b1903cd78ade5b10cb8b30830
T+16634ms:   Contract created: 0xe4395360de36b1905cb93d11e11ac24d04ed7d66
T+16634ms:   Gas usage: 1994320
T+16634ms:   Block Number: 3
T+16634ms:   Block Time: Mon Jun 24 2019 07:22:35 GMT+0100 (BST)
T+16648ms: eth_getBlockByNumber
T+16648ms: eth_getTransactionReceipt
T+16664ms: eth_getCode
T+16678ms: eth_getTransactionByHash
T+16689ms: eth_getBlockByNumber
T+16772ms: eth_getBalance
T+16772ms: eth_getBlockByNumber
T+16772ms: eth_getBlockByNumber
T+16772ms: eth_getBlockByNumber
T+16772ms: eth_estimateGas
T+16800ms: eth_getBlockByNumber
T+16829ms: eth_blockNumber
T+16857ms: eth_sendTransaction
T+16878ms:   Transaction: 0x21620449d4e362134537c671705b25210e29f02868b8bf26e21e56f7a5c984a8
T+16878ms:   Contract created: 0xeaa2720dc6d572b7716a684cc0d87f6e07061eae
T+16878ms:   Gas usage: 1372390
T+16878ms:   Block Number: 4
T+16905ms:   Block Time: Mon Jun 24 2019 07:22:35 GMT+0100 (BST)
T+16934ms: eth_getBlockByNumber
T+16962ms: eth_getTransactionReceipt
T+16991ms: eth_getCode
T+17011ms: eth_getTransactionByHash
T+17038ms: eth_getBlockByNumber
T+17064ms: eth_getBalance
T+17091ms: eth_getBlockByNumber
T+17111ms: eth_getBlockByNumber
T+17141ms: eth_getBlockByNumber
T+17168ms: eth_estimateGas
T+17199ms: eth_getBlockByNumber
T+17228ms: eth_blockNumber
T+17249ms: eth_sendTransaction
T+17280ms:   Transaction: 0x3dc8adb6261bcf1e5fba88dc532b0bddfe7c6e1364578b1ecdbcde951535ad05
T+17280ms:   Contract created: 0xf67c5535c3ef1661a06db6a6859e2aa51b14f7cf
T+17280ms:   Gas usage: 481467
T+17280ms:   Block Number: 5
T+17280ms:   Block Time: Mon Jun 24 2019 07:22:35 GMT+0100 (BST)
T+17309ms: eth_getBlockByNumber
T+17338ms: eth_getTransactionReceipt
T+17358ms: eth_getCode
T+17387ms: eth_getTransactionByHash
T+17417ms: eth_getBlockByNumber
T+17447ms: eth_getBalance
T+17479ms: eth_getBlockByNumber
T+17500ms: eth_getBlockByNumber
T+17532ms: eth_sendTransaction
T+17561ms:   Transaction: 0xd17ad2254aa29f4d79e519f059d9308e033c762f2c409c20130d466da3914334
T+17561ms:   Gas usage: 27023
T+17561ms:   Block Number: 6
T+17561ms:   Block Time: Mon Jun 24 2019 07:22:35 GMT+0100 (BST)
T+17590ms: eth_getBlockByNumber
T+17617ms: eth_getTransactionReceipt
T+21205ms: net_version
T+21222ms: eth_getCode
T+21236ms: eth_getBlockByNumber
T+21256ms: eth_getBalance
T+21270ms: eth_getTransactionCount
T+21287ms: eth_getStorageAt
T+21303ms: eth_getStorageAt
T+21319ms: eth_getStorageAt
T+21332ms: eth_getStorageAt
T+21348ms: eth_getStorageAt
T+21367ms: eth_getStorageAt
T+21382ms: eth_getStorageAt
T+21400ms: eth_getStorageAt
T+21414ms: eth_getStorageAt
T+21435ms: eth_getStorageAt
T+21448ms: eth_getStorageAt
T+21469ms: eth_getStorageAt
T+21483ms: eth_getStorageAt
T+21498ms: eth_getStorageAt
T+21517ms: eth_getStorageAt
T+21531ms: eth_getStorageAt
T+21553ms: eth_getStorageAt
T+21569ms: eth_getStorageAt
T+21588ms: eth_getStorageAt
T+21602ms: eth_getStorageAt
T+21620ms: eth_getStorageAt
T+37653ms: net_version
T+37676ms: eth_accounts
T+37723ms: eth_getBlockByNumber
T+37742ms: eth_getBlockByNumber
T+37757ms: eth_getBlockByNumber
T+37774ms: eth_sendTransaction
T+39060ms:   Transaction: 0xc452dc209d9437018822e3624af905a14601a3681f96b8063209e7aacb63d025
T+39060ms:   Gas usage: 299802
T+39060ms:   Block Number: 7
T+39078ms:   Block Time: Mon Jun 24 2019 07:23:06 GMT+0100 (BST)
T+39091ms: eth_getBlockByNumber
T+39110ms: eth_getTransactionReceipt
T+42997ms: net_version
T+43033ms: eth_unsubscribe
T+43047ms: eth_unsubscribe
T+43064ms: eth_unsubscribe
T+43077ms: eth_unsubscribe
T+43077ms: eth_unsubscribe
```

**Comment by alexandreValente** [comment]
Hello 

I had same issue, I solved it while removing everything inside $BASEDIR/build/contracts
Then I have migrated all the contracts again.

rm -rf build/contracts
truffle migrate --reset

FYI, I also tried before to restart Ganache, restart the workspace, Delete and recreate the workspace/local blockchain but nothing worked.
Therefore...remove anything from build/contract and truffle migrate again. (reset might be odd as build/contracts directory is empty).

NOTE: This can be acceptable while deploying in dev and test environment. Hope this will never happen in main network. I think it is a pure Ganache issue.
